tests/mocknvml/main.go

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+// busIdToString converts a [32]int8 array to a Go string
+func busIdToString(busId [32]int8) string {
+	bytes := make([]byte, 0, 32)
+	for _, b := range busId {
+		if b == 0 {
+			break
+		}
+		bytes = append(bytes, byte(b))
+	}
+	return strings.TrimRight(string(bytes), "\x00")
+}
+
+func main() {
+	log.Println("Starting Mock NVML Test")
+	log.Println("=================================")
+
+	// Initialize NVML
+	log.Println("Initializing NVML...")
```

Comment:
This is an integration test to check that we can use `go-nvml` with the mock `libnvidia-ml.so`